backend/functions/tests/flashcard.test.js

```diff
@@ -223,27 +235,136 @@ describe("Testing Flashcard Set Functions", () => {
     });
 
     test("throws error updating flashcard set we don't own", async () => {
-      // Create a test flashcard set for error cases
+      const context = {auth: {uid: user.uid}};
+      // Attempt to update flashcard set
+      const updateFlashcardSet = fft.wrap(myFunctions.updateFlashcardSet);
+      await expect(updateFlashcardSet({flashcardId: unownedFCSetId, ...unownedFCSetData}, context)).rejects.toEqual(
+          new Error("You do not have permission to edit this flashcard set."),
+      );
+
+      // Validate flashcard set still exists
+      const doc = await admin.firestore().collection("flashcards").doc(unownedFCSetId).get();
+      expect(doc.exists).toBeTruthy();
+    });
+  });
+
+  //* **************** Test deleteFlashcardSet ******************* */
+  describe("deleteFlashcardSet", () => {
+    let delDocId;
+
+    beforeAll(async () => {
       const data = {
-        creatorId: "some-uuid",
-        title: "Unowned Flashcard set",
+        creatorId: user.uid,
+        title: "Doc to delete",
         category: "Test",
         timestamp: Date.now(),
         cards: [{question: "Q", answer: "A"}],
       };
       const docRef = await admin.firestore().collection("flashcards").add(data);
-      docsToDelete.push(docRef.id); // For cleanup
+      delDocId = docRef.id;
+      docsToDelete.push(delDocId); // For cleanup
+    });
 
+    test(`"trash" doc in /flashcards`, async () => {
       const context = {auth: {uid: user.uid}};
-      // Attempt to update flashcard set
-      const updateFlashcardSet = fft.wrap(myFunctions.updateFlashcardSet);
-      await expect(updateFlashcardSet({flashcardId: docRef.id, ...data}, context)).rejects.toEqual(
-          new Error("You do not have permission to edit this flashcard set."),
+      // "Trash" Flashcard
+      const deleteFlashcardSet = fft.wrap(myFunctions.deleteFlashcardSet);
+      const res = await deleteFlashcardSet({flashcardId: delDocId}, context);
+      expect(res.message).toEqual("The flashcard set was sent to the trash.");
+
+      // Valdiate flashcard set still exists, but with "toBeDeleted = true"
+      const doc = await admin.firestore().collection("flashcards").doc(delDocId).get();
+      expect(doc.exists).toBeTruthy();
+      expect(doc.data().toBeDeleted).toBe(true);
```

Comment:
I think we need to add `toBeDeleted` to the metadata in the helper function `formatFlashcardSetData `. in order to display `createdFlashcard` to the user and `deletedFlashcard` in the trash. we need to be able to filter by `toBeDeleted`

Reply:
Otherwise. the code looks good!

Reply:
> I think we need to add `toBeDeleted` to the metadata in the helper function `formatFlashcardSetData `.

It should be fine since we don't reference it in the delete or recover functions. The tests directly fetches from the database to check for the `toBeDeleted` property.


> In order to display `createdFlashcard` to the user and `deletedFlashcard` in the trash. we need to be able to filter by `toBeDeleted`

I think since we'll probably fetch data on the client-side, we can just use `where` to filter search results.